databuilder/extractor/neo4j_metric_search_data_extractor.py

```diff
@@ -0,0 +1,86 @@
+import textwrap
+from typing import Any  # noqa: F401
+
+from pyhocon import ConfigTree  # noqa: F401
+
+from databuilder import Scoped
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.neo4j_extractor import Neo4jExtractor
+from databuilder.publisher.neo4j_csv_publisher import JOB_PUBLISH_TAG
+
+
+class Neo4jMetricSearchDataExtractor(Extractor):
+    """
+    Extractor to fetch data required to support search from Neo4j graph database
+    Use Neo4jExtractor extractor class
+    """
+    CYPHER_QUERY_CONFIG_KEY = 'cypher_query'
+
+    DEFAULT_NEO4J_CYPHER_QUERY = textwrap.dedent(
+        """
+        MATCH (m)-[:METRIC_OF]->(d:Dashboard)
```

Comment:
Relation explained in main thread. As I know Superset has metrics as well, but they are "hidden" behind what they call charts in Superset. We understand the metric as a formula (expression) how the data are calculated for the dashboard (chart). 

The model in Qlik Sense is more or less similar to Superset. In Superset, there is Dashboard - Chart - and chart has a definition (metric). Qlik Sense has Stream - Dashboard - chart - metric. But chart is just a box which defines a chart type, visualization properties, name, colors, what metric and dimension to use etc. and in our case the most users ask us "where can I find  ... invoicing from the last year" and they want the answer "Finance dashboard in stream finance", if you have privileges :). And invoicing is a metric for us. So I decided to skip chart as we use something called `Master metrics` - you can use them in the whole dashboard in more charts and at the end we would like to have something like `Master metrics repository` - outside of Qlik and one metric with the same id should be calculated same way in the whole company. The differences between Qlik Sense and Superset are in ownerships of objects. In Superset you have an owner of Dashboard and you have an owner of chart. In Qlik Sense have just the owner of dashboard. And maybe, but I don't remember it and I can't check it at the moment - in Superset you can run reload/refresh of chart, but in Qlik Sense there is a reload function only for whole dashboard. But if I think about Superset, it could be the same model for Superset and Qlik Sense, if we add chart level in hierarchy.  I am sorry, I have last 5 % of battery and I am without electricity until Sunday late evening, so the rest of answers on Monday.